doc/sphinx/proof-engine/tactics.rst

```diff
@@ -2521,12 +2521,28 @@ and an explanation of the underlying technique.
    :g:`H : x = y` and a hypothesis of type :g:`rew H in p = q`); these
    hypotheses can subsequently be simplified using :tacn:`subst`, without ever
    invoking any kind of axiom asserting uniqueness of identity proofs. If you
-   want to explicitly specify the hypothesis to be inverted, or name the
+   want to explicitly specify the hypothesis to be inverted, you can pass it as
+   an argument to :tacn:`inversion_sigma`. If you want to name the
    generated hypotheses, you can invoke
    :n:`induction H as [H1 H2] using eq_sigT_rect.` This tactic also works for
-   :g:`sig`, :g:`sigT2`, and :g:`sig2`, and there are similar :g:`eq_sig`
+   :g:`sig`, :g:`sigT2`, :g:`sig2`, :g:`ex`, and :g:`ex2`, and there are similar :g:`eq_sig`
    :g:`***_rect` induction lemmas.
 
+
+   .. exn:: Type of @ident is not an equality of recognized Σ types: expected one of sig , sig2 , sigT , sigT2 , sigT2 , ex , ex2 but got @term
```

Comment:
`sig` and the others are basically literal text, modulo a kludge I'm using to display appropriately qualified names.  (If you pass `-noinit` and then do `Require Coq.Init.Tactics`, then the error message will display appropriately qualified versions of `sig`, etc.  Also similarly if you define `Definition sig := ...`.)  This same kludge inserts spaces before the commas, and I'm not sure how to avoid it while still getting the appropriately qualified versions of the constants in the error messages.

Reply:
> `sig` and the others are basically literal text

I don't follow you.  When I look at the code producing the message, they are variables that are printed.  Can you show me the message from an example?  Maybe then I will get it.

If Coq prints the extra spaces, let's keep them in the message so it's easier to search for in the doc.

Reply:
> I don't follow you. When I look at the code producing the message, they are variables that are printed. Can you show me the message from an example? Maybe then I will get it.

The code is essentially equivalent to `let sig := uconstr:(@sig) in in idtac sig`.  Here is a simpler example with the same principle:
```coq
Module A.
  Definition foo := I.
  Ltac bar := let f := uconstr:(foo) in idtac "foo qualified is" f.
End A.

Goal True.
  A.bar. (* foo qualified is A.foo *)
  Import A.
  bar. (* foo qualified is foo *)
```
Here is an example with the actual tactic:
```coq
Unset Ltac Backtrace.
Goal True = True -> True.
  intro H.
  inversion_sigma H.
  (* Error: Tactic failure: Type of H
is not an equality of recognized Σ types: expected one of sig , sig2 , sigT ,
sigT2 , ex , ex2 but got Prop. *)
```

Reply:
> I don't follow you. When I look at the code producing the message, they are variables that are printed. Can you show me the message from an example? Maybe then I will get it.

The code is essentially equivalent to `let sig := uconstr:(@sig) in in idtac sig`.  Here is a simpler example with the same principle:
```coq
Module A.
  Definition foo := I.
  Ltac bar := let f := uconstr:(foo) in idtac "foo qualified is" f.
End A.

Goal True.
  A.bar. (* foo qualified is A.foo *)
  Import A.
  bar. (* foo qualified is foo *)
```
Here is an example with the actual tactic:
```coq
Unset Ltac Backtrace.
Goal True = True -> True.
  intro H.
  inversion_sigma H.
  (* Error: Tactic failure: Type of H
is not an equality of recognized Σ types: expected one of sig , sig2 , sigT ,
sigT2 , ex , ex2 but got Prop. *)
```

Reply:
>This same kludge inserts spaces before the commas, and I'm not sure how to avoid it while still getting the appropriately qualified versions of the constants in the error messages.

Instead of comma-separated you could do space-separated, ie
```suggestion
   .. exn:: Type of @ident is not an equality of recognized Σ types: expected one of sig sig2 sigT sigT2 sigT2 ex or ex2 but got @term
```

Reply:
Do you think that's better?

Reply:
Done